### ISSUE TITLE
Bower dependencies causing Che build to fail

### ISSUE THREAD

**Issue by helderm** [eclipse-che/che#1827]
While performing a full build on Che through Docker, I'm getting errors during the execution of `che-dashboard-war` project caused by Bower failing to get a status on some projects hosted on Github. 

**Reproduction Steps:**
1. Execute the build command through Docker

``` bash
$ docker run -it --rm --name build-che -v "$HOME/.m2:/home/user/.m2" -v "$PWD":/home/user/che-build -w /home/user/che-build codenvy/che-dev mvn -DskipTests=true -Dgwt.compiler.localWorkers=2 -T 1C -Dskip-validate-sources clean install
```

**Expected behavior:**

Succesfully compiled binaries

**Observed behavior:**
The build is failing during the Bower installation phase of some packages. The package itself seems to be different in every run, which would suggest that a timeout error may have caused Bower to crash.

``` bash
[exec] bower card#>=0.1.4                   checkout v2.0.0
[exec] bower angular#~1.3.8                 progress remote: Compressing objects: 100% (10/10), done.
[exec] bower angular#~1.3.8                 resolved https://github.com/angular/bower-angular.git#1.3.20
[exec] bower angular-moment#0.9.0            ECMDERR Failed to execute "git ls-remote --tags --heads https://github.com/urish/angular-moment.git", exit code of #128 fatal: unable to access 'https://github.com/urish/angular-moment.git/': Empty reply from server
[exec] 
[exec] Additional error details:
[exec] fatal: unable to access 'https://github.com/urish/angular-moment.git/': Empty reply from server
```

**Che version:**    SHA 0d16da4ee0760dd672a5ceba837f8bd688542cd2
**OS and version:** Linux trusty 4.2.0-42-generic #49~14.04.1-Ubuntu SMP Wed Jun 29 20:22:11 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
**Docker version:** Docker version 1.11.2, build b9f10c9
**Che install:**   Docker container
**Container version:** codenvy/che-dev     latest              1f9ad2e2af4d        5 weeks ago         2 GB

**Additional information:**
- Problem started happening recently, didn't happen in an older version of Che: Yes
- Problem can be reliably reproduced, doesn't happen randomly: Yes

**Comment by ddementieva** [comment]
@helderm Can you check if https://github.com/urish/angular-moment.git repo is available for you?

**Comment by ghost** [comment]
@helderm I just worked for me. Make sure you have got no connectivity issues on your host machine and in a container.

**Comment by helderm** [comment]
The repo was available from inside the container the moment that I tested, and I had no connectivity issues as can be seen in the log from the other packages installed during the same session. I don't discard that connection may have been slow during build, which may have caused a timeout while installing that package.

I tested it again today, and the build is going fine for me as well. Anyway, I think it would be a plus if the build process were more resilient against this kind of problem. I researched a bit and noticed that Bower timeout can be configured in `.bowerrc` files by setting the `timeout` property, which defaults to `60000ms` today. None of the `.bowerrc` files in the repo explicitly sets the timeout, maybe adding a property during build that customizes this value would be beneficial.
